tests/subsys/logging/log_syst/src/main.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Test log message
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include "mock_backend.h"
+#include <sys/printk.h>
+#include <logging/log_backend.h>
+#include <logging/log_backend_std.h>
+#include <logging/log_ctrl.h>
+#include <logging/log_output.h>
+#include <tc_util.h>
+#include <stdbool.h>
+#include <ztest.h>
+#include <stdlib.h>
+
+#define TEST_MSG_0 "0 args"
+#define PAYLOAD_MULTIPLE_ARGS "4465627567206D657373616765206578616D706C652C2025642C2025642C20256"\
+	"400010000000200000003000000"
+
+MOCK_LOG_BACKEND_DEFINE(backend1, false);
+
+#define LOG_MODULE_NAME test
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+/**
+ * @brief Testcase to validate that the mock backend would use the expected
+ * processing function from the function pointer table format_table
+ */
+void test_log_syst_format_table_selection(void)
+{
+
+#ifdef CONFIG_LOG_MIPI_SYST_ENABLE
+	uint32_t test_log_type_syst = LOG_OUTPUT_SYST;
+
+	log_format_func_t test_log_output_func = log_format_func_t_get(test_log_type_syst);
+
+	zassert_equal_ptr(test_log_output_func, log_output_msg2_syst_process,
+					"Correct Function pointer for SYST log\n"
+					"format was not selected %p vs %p",
+					test_log_output_func, log_output_msg2_syst_process);
+
+#elif CONFIG_LOG_BACKEND_MOCK_OUTPUT_DEFAULT == LOG_OUTPUT_TEXT
+
+	uint32_t test_log_type_text = LOG_OUTPUT_TEXT;
+	log_format_func_t test_log_output_func = log_format_func_t_get(test_log_type_text);
+
+	zassert_equal_ptr(test_log_output_func, log_output_msg2_process,
+						"Function pointer for TEXT log format was not selected");
+
+#endif
+
+}
+
+#ifdef CONFIG_LOG_MIPI_SYST_ENABLE
+/* Testcase to validate the SYST output of log data */
+void test_log_syst_data(void)
+{
+	LOG_DBG("Debug message example, %d", 1);
+
+	const char *type = "72";
+	const char *optional_flags = "0A";
+	const char *module_id = "00";
+	const char *sub_type;
+#ifdef CONFIG_64BIT
+	sub_type = "0C";
+#else
+	sub_type = "0B";
+#endif
+	const char *payload =
+	"4465627567206D657373616765206578616D706C652C2025640001000000";
```

Comment:
This comment is a maintenance suggestion. Feel free to ignore.

I note that the payload in each of the tests is comprised of two hex-string parts--format string and arguments. As the hex-strings are somewhat harder to interpret, it may be worthwhile to have an intermediary validation routine that converts a string to a hex-string. Not only should this make it easier to follow what each payload test is verifying, it also provides some flexibility should the format strings change in the future.